src/test/shell/unittest.bash

```diff
@@ -597,6 +597,37 @@ function assert_not_contains() {
     return 1
 }
 
+# Usage: assert_consecutive_lines_in_file <file> <line_1> <line_2> ... <line_n>
+# Asserts that file contains the given lines in the given order and returns zero
+# or non-zero accordingly.
+# It assumes the first line is unique in the given file. If the first line is
+# not unique it may return a wrong answer.
+function assert_consecutive_lines_in_file() {
```

Comment:
This function it's too complex for a test setup.

I suggest to use some form of grep or awk:
https://stackoverflow.com/questions/43398350/grep-bash-multi-line-pattern

Or ultimately use something like
 tr '\n' , < foo.txt | grep -o "begin.*end" | tr , '\n'

To remove the newlines, pattern match then restore them.

Reply:
Is there a guideline on how complex a test setup should be? What specifically do you find complex here? Given both versions do the same thing and they would be properly documented either one of them should do. I'm more inclined towards the iterative version because: 
- I find it more readable and clear since it is closer to code written in other languages (unfortunately bazel developers are not bash gurus)
- The inputs are clearly defined and don't require any mangling (e.g. removing and restoring newlines); thinking again I don't think it's a good idea to alter the input file at all
- Finding _and_ understanding the right combination of flags and chained commands requires more time from the person who writes and reads this method

Reply:
I ended up using tr and grep. I added comments explaining the code. Even though I feel more comfortable with the Java-like version, it was more error prone in the end. _sigh_ 
I also realized that your suggestion doesn't alter the input file like I initially thought. Now I think the tr & grep version satisfies my 3 grunts from above. :)